First part of Homework 5 done

MegaLexa11 opened the pull request:
Выполнил только первую часть

MegaLexa11 committed:
First part of Homework 5 done

Выполнил только первую часть

avchirkin commented on Homework_5/Homework_5/ProductItem.cs:
```diff
@@ -0,0 +1,29 @@
+    // Сущность товара
+    internal class ProductItem
+    {
+        private Guid Id;
```
Лучше сделать публичным свойством, так как на фронтенд и обратно ты будешь прикидывать эти сущности в разрезе ID. По имени нельзя - у разных товаров в теории может быть одно и то же название.

avchirkin commented on Homework_5/Homework_5/Stock.cs:
```diff
@@ -0,0 +1,47 @@
+        // Подумать еще над этой штукой
+        public List<ProductItem> Items { get; }
+        
+        public Stock(int baseSize)
```
"Вместимость" склада не очень вяжется с реальным ассортиментом - как её определить? А главное - зачем?

avchirkin commented on Homework_5/Homework_5/Stock.cs:
```diff
@@ -0,0 +1,47 @@
+        // Проверить
+        public void AddToStock(ProductItem item)
+        {
+            if (!Items.Contains(item))
```
У тебя при случайном добавлении одного и того же товара два раза будет генериться новый ID в ProductItem, поэтому эта проверка никогда не сработает - от дублей ты так не защитишься

avchirkin commented on Homework_5/Homework_5/Stock.cs:
```diff
@@ -0,0 +1,47 @@
+            }
+        }
+
+        public ProductItem this[int index]
```
Зачем здесь индексатор? Если нам нужно будет получить какой-то продукт со склада, то делаться это будет, скорее всего, по имени, категории или ID

avchirkin commented on Homework_5/Homework_5/ProductInStorageMonitoring.cs:
```diff
@@ -0,0 +1,83 @@
+
+        private ProductInStorageMonitoring()
+        {
+            ItemStorageList = new List<ProductInStorage>(100);
```
Почему именно 100?

avchirkin commented on Homework_5/Homework_5/ProductInStorageMonitoring.cs:
```diff
@@ -0,0 +1,83 @@
+
+            amount = amount ?? 1;
+
+            if (itemExists)
```
Почему и правда не сделать через словарь, где ключ - например, ID товара, а значение - количество товаров на складе?

avchirkin commented on Homework_5/Homework_5/ProductInStorageMonitoring.cs:
```diff
@@ -0,0 +1,83 @@
+            }
+        }
+
+        // Шото с чем то, но пускай будет
```
Если ты не уверен в необходимости наличия кода, то его лучше удалять, иначе код превратится в "свалку идей", большая часть из которых вряд ли будет использована в будущем

avchirkin commented on Homework_5/Homework_5/Invoice.cs:
```diff
@@ -0,0 +1,36 @@
+            StorageDestination = storageDestination;
+        }
+
+        public void AddToInvoice(ProductItem item, int amount)
```
И так понятно, что мы добавляем в Invoice. Поэтому название метода можно упростить до, например, AddProduct или AddItem

avchirkin commented on Homework_5/Homework_5/ProductInStorage.cs:
```diff
@@ -0,0 +1,28 @@
+{
+    internal class ProductInStorage
+    {
+        public ProductItem Item;
```
Публичные поля лучше не использовать

avchirkin commented on Homework_5/Homework_5/ProductInStorage.cs:
```diff
@@ -0,0 +1,28 @@
+
+namespace Homework_5
+{
+    internal class ProductInStorage
```
Сама сущность - это, по сути, остаток на складе. Лучше переименовать соответствующим образом.

avchirkin reviewed (changes requested):
Понакидал замечаний

MegaLexa11 committed:
Removed all unnecessary stuff

Убрал два лишних класса, остатки теперь хранятся полем в рамках Storage, соответствующим образом были адаптированы к этому и другие классы, добавлена цена и ее история. Задачу по свойствам товаров не делал, думаю, что, если и сделаю, то в рамках следующего дз, там она, как будто, поуместнее

avchirkin commented on Homework_5/Homework_5/ProductItem.cs:
```diff
@@ -0,0 +1,32 @@
+    // Сущность товара
+    internal class ProductItem
+    {
+        public Guid Id;
```
Свойство

avchirkin commented on Homework_5/Homework_5/ProductItem.cs:
```diff
@@ -0,0 +1,32 @@
+
+        public string Measure { get; }
+
+        public ItemPrice Price;
```
Свойство

avchirkin commented on Homework_5/Homework_5/Storage.cs:
```diff
@@ -0,0 +1,58 @@
+    // Сущность склада
+    internal class Storage
+    {
+        public Guid Id;
```
Свойство

avchirkin commented on Homework_5/Homework_5/Storage.cs:
```diff
@@ -0,0 +1,58 @@
+            Stock = stock;
+        }
+
+        public void addItem(ProductItem item, int? amount = null)
```
Название метода - с большой буквы

avchirkin commented on Homework_5/Homework_5/ItemPrice.cs:
```diff
@@ -0,0 +1,35 @@
+
+        public int Price { get; set; }
+
+        Dictionary<DateTime, int> PriceStory = new Dictionary<DateTime, int>();
```
Добавить модификатор доступа

avchirkin reviewed (approved):
Аппрувнул, но поправь мелкие замечания

MegaLexa11 committed:
Some fixes
